services/ui-src/src/components/modals/DeleteDynamicFieldModal.tsx

```diff
@@ -0,0 +1,64 @@
+// components
+import { Text } from "@chakra-ui/react";
+import { Spinner } from "@cmsgov/design-system";
+import { Modal } from "components";
+import { useState } from "react";
+// types
+import { EntityShape, EntityType } from "types";
+
+export const DeleteDynamicFieldModal = ({
+  selectedRecord,
+  removeRecord,
+  entityType,
+  modalDisclosure,
+}: Props) => {
+  const [deleting, setDeleting] = useState<boolean>(false);
+  const convertToReadableEntityName = (name: any) => {
+    const fieldTypeMap = {
+      plans: "Plan",
+      bssEntities: "BSS Entity",
+    };
+    const fieldType: keyof typeof fieldTypeMap = name;
+    return fieldTypeMap[fieldType];
```

Comment:
Info: If you specify that `fieldTypeMap` has type `:any`, then you can save yourself the keyof typeof hassle.

```suggestion
    const fieldTypeMap: any = {
      plans: "Plan",
      bssEntities: "BSS Entity",
    };
    return fieldTypeMap[name];
```

Reply:
Thought: In fact, you could simplify things a bit in this file if you just declare the `fieldTypeMap` outside on like line 8, delete this function, then just reference the fieldTypeMap wherever you need it, like: `${fieldTypeMap[name]}`. More on this below.